lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/job/SwapUpdated.kt

```diff
@@ -92,27 +76,49 @@ class SwapUpdatedJob(
             .fold(StringBuilder()) { sb, it -> sb.append("%02x".format(it)) }
             .toString()
 
+    private fun handlePaymentEvent(payment: Payment) {
+        val swapId = when (val details = payment.details) {
+            is PaymentDetails.Bitcoin -> details.swapId
+            is PaymentDetails.Lightning -> details.swapId
+            else -> null
+        }
+
+        swapId?.let {
+            if (this.swapIdHash == hashId(it)) {
+                logger.log(
+                    TAG,
+                    "Received payment event: ${this.swapIdHash} ${payment.status}",
+                    "TRACE"
+                )
+                notifySuccess(payment)
+            }
+        }
+    }
+
     private fun notifySuccess(payment: Payment) {
-        logger.log(TAG, "Payment ${payment.txId} completed successfully", "INFO")
-        val received = payment.paymentType == PaymentType.RECEIVE
-        notifyChannel(
-            context,
-            NOTIFICATION_CHANNEL_SWAP_UPDATED,
-            getString(
+        if (!this.notified) {
+            logger.log(TAG, "Payment ${payment.txId} processing successful", "INFO")
+            val received = payment.paymentType == PaymentType.RECEIVE
+            notifyChannel(
                 context,
-                if (received) PAYMENT_RECEIVED_NOTIFICATION_TITLE else PAYMENT_SENT_NOTIFICATION_TITLE,
-                if (received) DEFAULT_PAYMENT_RECEIVED_NOTIFICATION_TITLE else DEFAULT_PAYMENT_SENT_NOTIFICATION_TITLE
-            ),
-            String.format(
+                NOTIFICATION_CHANNEL_SWAP_UPDATED,
                 getString(
                     context,
-                    if (received) PAYMENT_RECEIVED_NOTIFICATION_TEXT else PAYMENT_SENT_NOTIFICATION_TEXT,
-                    "%d",
-                    if (received) DEFAULT_PAYMENT_RECEIVED_NOTIFICATION_TEXT else DEFAULT_PAYMENT_SENT_NOTIFICATION_TEXT
-                ), payment.amountSat.toLong()
+                    if (received) PAYMENT_RECEIVED_NOTIFICATION_TITLE else PAYMENT_SENT_NOTIFICATION_TITLE,
+                    if (received) DEFAULT_PAYMENT_RECEIVED_NOTIFICATION_TITLE else DEFAULT_PAYMENT_SENT_NOTIFICATION_TITLE
+                ),
+                String.format(
+                    getString(
+                        context,
+                        if (received) PAYMENT_RECEIVED_NOTIFICATION_TEXT else PAYMENT_SENT_NOTIFICATION_TEXT,
+                        "%d",
+                        if (received) DEFAULT_PAYMENT_RECEIVED_NOTIFICATION_TEXT else DEFAULT_PAYMENT_SENT_NOTIFICATION_TEXT
+                    ), payment.amountSat.toLong()
+                )
             )
-        )
-        fgService.onFinished(this)
+            this.notified = true
```

Comment:
Rather than introducing a new boolean flag, consider resetting `swapIdHash` to null once a notification is sent.